### ISSUE TITLE
grpo显存爆炸相关问题

### ISSUE THREAD

**Issue by Obsever003** [jingyaogong/minimind#827]
model_unwrapped = model.module if isinstance(model, DistributedDataParallel) else model
        with autocast_ctx:
            res = model_unwrapped(outputs, attention_mask=full_mask)
            aux_loss = res.aux_loss if lm_config.use_moe else torch.tensor(0.0, device=args.device)
            per_token_logps = F.log_softmax(res.logits[:, :-1, :], dim=-1).gather(2, outputs[:, 1:].unsqueeze(-1)).squeeze(-1).gather(1, logp_pos)
观察到64m模型在这一段代码中显存消耗暴涨20gb；这么大的显存消耗是什么原因？

**Comment by jingyaogong** [comment]
- Context 最长 1792 tokens：768 prompt + 1024 生成
- 一次前向 12 个样本：batch 2 × 每题生成 6 条
- Attention 矩阵大小是 `[12, 8, 1792, 1792]`。有 padding 时不走 Flash Attention，8 层普通 Attention 实测占 14.8GB
- +RMSNorm 4.1GB、MLP 3.1GB、logits 和 `log_softmax` 约 1.3GB，总峰值约 23.4GB
